www/js/metrics/CarbonFootprintCard.tsx

```diff
@@ -0,0 +1,168 @@
+import { formatDateRangeOfDays, parseDataFromMetrics, generateSummaryFromData, calculatePercentChange, segmentDaysByWeeks, isCustomLabels } from './metricsHelper';
+import { useTranslation } from 'react-i18next';
+import BarChart from '../components/BarChart';
+import { getAngularService } from '../angular-react-helper';
+import ChangeIndicator from './ChangeIndicator';
+import color from "color";
+
+type Props = { userMetrics: MetricsData, aggMetrics: MetricsData }
+const CarbonFootprintCard = ({ userMetrics, aggMetrics }: Props) => {
+    const FootprintHelper = getAngularService("FootprintHelper");
+    const { colors } = useTheme();
+    const { t } = useTranslation();
+
+    const [emissionsChange, setEmissionsChange] = useState({});
+ 
+    const userCarbonRecords = useMemo(() => {
+        if(userMetrics?.distance?.length > 0) {
+            //separate data into weeks
+            const [thisWeekDistance, lastWeekDistance] = segmentDaysByWeeks(userMetrics?.distance, 2);
+
+            //formatted data from last week
+            let userLastWeekModeMap = {};
+            let userLastWeekSummaryMap = {};
+            if(lastWeekDistance && lastWeekDistance?.length >= 7) {
+                userLastWeekModeMap = parseDataFromMetrics(lastWeekDistance, 'user');
+                userLastWeekSummaryMap = generateSummaryFromData(userLastWeekModeMap, 'distance');
+            }
+            
+            //formatted distance data from this week
+            let userThisWeekModeMap = parseDataFromMetrics(thisWeekDistance, 'user');
+            let userThisWeekSummaryMap = generateSummaryFromData(userThisWeekModeMap, 'distance');
+            let worstDistance = userThisWeekSummaryMap.reduce((prevDistance, currModeSummary) => prevDistance + currModeSummary.values, 0);
+            
+            //setting up data to be displayed
+            let graphRecords = [];
+
+            //set custon dataset, if the labels are custom
+            if(isCustomLabels(userThisWeekModeMap)){
+                FootprintHelper.setUseCustomFootprint();
+            }
+
+            //calculate low-high and format range for prev week, if exists
+            let userPrevWeek;
+            if(userLastWeekSummaryMap[0]) {
+                userPrevWeek = {
+                    low: FootprintHelper.getFootprintForMetrics(userLastWeekSummaryMap, 0), 
+                    high: FootprintHelper.getFootprintForMetrics(userLastWeekSummaryMap, FootprintHelper.getHighestFootprint())
+                };
+                graphRecords.push({label: t('main-metrics.unlabeled'),  x: userPrevWeek.high - userPrevWeek.low, y: `${t('main-metrics.prev-week')}\n(${formatDateRangeOfDays(lastWeekDistance)})`})
+                graphRecords.push({label: t('main-metrics.labeled'), x: userPrevWeek.low, y: `${t('main-metrics.prev-week')}\n(${formatDateRangeOfDays(lastWeekDistance)})`});
+            }
+
+            //calculate low-high and format range for past week
```

Comment:
can we clarify that past week = past 7 days? Maybe this is just my verbal blind spot, but whenever I see "past week", I think that it is the last week, not the "current week". And since this is code, we have a lot of space for clarification.

Reply:
I've just added some comments to this section, referring to "how many days ago" the calculations are referencing, so the previous week is 14 days ago - 8 days ago, and the past week is 7 days ago - yesterday. Depending on user selection, this could be shifted up to include "today".